sdk/pinocchio/src/entrypoint/middleware.rs

```diff
@@ -0,0 +1,35 @@
+//! Defines the middleware entrypoint, enabling a hot path to bypass
+//! entrypoint deserialization, ejecting to the cold path on failure.
+#[macro_export]
+macro_rules! middleware_entrypoint {
```

Comment:
Adding a "program" prefix since this macro does not include the allocator and panic handler.

```suggestion
macro_rules! middleware_program_entrypoint {
```